src/main/app/forms/validation/validators/amountIsDefined.ts

```diff
@@ -0,0 +1,33 @@
+import {
+  registerDecorator,
+  ValidationArguments,
+  ValidationOptions,
+  ValidatorConstraint,
+  ValidatorConstraintInterface
+} from 'class-validator'
+
+@ValidatorConstraint({ name: 'amountIsDefined' })
+export class AmountIsDefinedConstraint implements ValidatorConstraintInterface {
+
+  validate (value: any, args: ValidationArguments) {
+    const cannotStateInput = (args.object as any)[args.constraints[0]]
+
+    if (cannotStateInput === 'cannot' && (value === undefined || value === null || isNaN(value))) {
+      return true
+    }
+
+    return value != null
+  }
+}
+
+export function AmountIsDefined (cannotState: string, validationOptions?: ValidationOptions) {
```

Comment:
As this validator can be used with specific type of amount, i.e. AmountRange. Probably rename it accordingly. Adding some java docs will be helpful as well